punchcut/src/scaler.rs

```diff
@@ -122,35 +123,43 @@ impl<'a> ScalerBuilder<'a> {
         };
         let avar_mappings = font.avar().ok().map(|avar| avar.axis_segment_maps());
         let axis_count = fvar.axis_count() as usize;
+        self.context.coords.clear();
         self.context
             .coords
             .resize(axis_count, NormalizedCoord::default());
-
-        for (i, (axis, dest_coord)) in axes.iter().zip(&mut self.context.coords).enumerate() {
-            let tag = axis.axis_tag();
-            for variation in &self.context.variations {
-                if variation.tag == tag {
-                    let mut coord = axis.normalize(Fixed::from_f64(variation.value as f64));
-                    coord = avar_mappings
-                        .as_ref()
-                        .and_then(|mappings| mappings.get(i).transpose().ok())
-                        .flatten()
-                        .map(|mapping| mapping.apply(coord))
-                        .unwrap_or(coord);
-                    let coord = coord.to_f64() as f32;
-                    *dest_coord = NormalizedCoord::from_f32(coord);
-                }
+        for variation in &self.context.variations {
+            // Make sure we iterate over all axes to support the case with multiple axes
+            // with the same name.
```

Comment:
Suggest expanding to note this permits non-linear interpolation. Could cite https://github.com/PeterConstable/OT_Drafts/blob/master/NLI/UnderstandingNLI.md.

Reply:
Thanks for the link. Added.